memory/node.go

```diff
@@ -187,3 +188,91 @@ func LinkCast(v interface{}) (l Link, ok bool) {
 	}
 	return l, true
 }
+
+func (n Node) Read(fun stream.ReadFun) error {
+	err := read(n, fun, []interface{}{})
+	if err == stream.NodeReadAbort {
+		err = nil
+	}
+	return err
+}
+
+func read(curr interface{}, fun stream.ReadFun, path []interface{}) error {
```

Comment:
I think this may be difficult. Also, I don't like abstractions just for the sake of them. Sometimes code is easier to read with a little repetition.

But still, this may be needing more thought

- [ ] rework `memory.Node.Read`